File: csv/cplus/create/csv_create.cpp
// --------------------------------------------------------------------
/*
	csv_create.cpp

					Jan/30/2015

*/
// --------------------------------------------------------------------
#include	<iostream>
#include	<fstream>
#include	<map>

#include	<string.h>
#include	<boost/lexical_cast.hpp>

using namespace std;
using namespace boost;

typedef map<string,string> Unit;

// --------------------------------------------------------------------
extern	void dict_display_proc (map <string,Unit> dict_aa);
extern	map <string,Unit> dict_append_proc
	(map <string,Unit> dict_aa,string id,string name,
	int population,string date_mod);

extern	void csv_write_proc (char file_out[],map <string,Unit> dict_aa);

// --------------------------------------------------------------------
static map <string,Unit >  data_prepare_proc ()
{
	map <string,Unit> dict_aa;

	dict_aa = dict_append_proc (dict_aa,"t1271","千葉",75196,"1922-8-17");

	dict_aa = dict_append_proc (dict_aa,"t1272","勝浦",13682,"1922-9-8");
	dict_aa = dict_append_proc (dict_aa,"t1273","市原",47521,"1922-7-25");
	dict_aa = dict_append_proc (dict_aa,"t1274","流山",29817,"1922-6-14");
	dict_aa = dict_append_proc (dict_aa,"t1275","八千代",54371,"1922-1-5");
	dict_aa = dict_append_proc (dict_aa,"t1276","我孫子",76148,"1922-5-22");
	dict_aa = dict_append_proc (dict_aa,"t1277","鴨川",46725,"1922-7-17");
	dict_aa = dict_append_proc (dict_aa,"t1278","銚子",52614,"1922-3-04");
	dict_aa = dict_append_proc (dict_aa,"t1279","市川",74893,"1922-5-9");

	return	dict_aa;
}

// --------------------------------------------------------------------
int main (int argc,char *argv[])
{
	cout << "*** 開始 ***\n";

	char file_out[160];

	strcpy (file_out,argv[1]);

//	cerr << "\tfile_out = " << file_out << "\n";

	map <string,Unit> dict_aa = data_prepare_proc ();

	cout << "要素数：" << (unsigned int)dict_aa.size() << endl;

	dict_display_proc (dict_aa);

	csv_write_proc (file_out,dict_aa);

	cout << "*** 終了 ***\n";
}

// --------------------------------------------------------------------

File: csv/vb/create/csv_create.vb
'------------------------------------------------------------------------
'	csv_create.vb
'
'						Apr/07/2015
'
'------------------------------------------------------------------------
Imports System.IO
Imports System.Collections

Module Module_text
'------------------------------------------------------------------------
Function data_prepare_proc () as Hashtable
	Dim dict_aa As New Hashtable()

	dict_aa ("t1271") = unit_gen_proc ("千葉","46175","1959-9-24")
	dict_aa ("t1272") = unit_gen_proc ("勝浦","82536","1959-1-18")
	dict_aa ("t1273") = unit_gen_proc ("市原","92365","1959-5-9")
	dict_aa ("t1274") = unit_gen_proc ("流山","15823","1959-7-15")
	dict_aa ("t1275") = unit_gen_proc ("八千代","97241","1959-6-24")
	dict_aa ("t1276") = unit_gen_proc ("我孫子","72614","1959-4-12")
	dict_aa ("t1277") = unit_gen_proc ("鴨川","53472","1959-7-18")
	dict_aa ("t1278") = unit_gen_proc ("銚子","65814","1959-5-21")
	dict_aa ("t1279") = unit_gen_proc ("市川","28951","1959-1-20")

	return	dict_aa
End Function
'------------------------------------------------------------------------

Sub Main()
	Dim clArgs() As String = Environment.GetCommandLineArgs()
	console.writeLine ("*** 開始 ***")
	Dim file_text As String
	file_text = clArgs(1)
	console.writeLine (file_text)

	Dim dict_aa as HashTable =  data_prepare_proc ()
	dict_to_file_proc (dict_aa,file_text,",")
 
	console.writeLine ("*** 終了 ***")
End Sub

'------------------------------------------------------------------------
End Module

'------------------------------------------------------------------------
